test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java

```diff
@@ -84,14 +88,15 @@ public static void main(String[] args) throws Exception {
         Thread thread2 = builder2.start(() -> {
             synchronized (LOCK2) {
                 try { barrier.await(); } catch (Exception ie) {}
+                reached2.set(true);
                 synchronized (LOCK1) { }
             }
         });
         System.out.println("thread2 => " + thread2);
 
         System.out.println("Waiting for thread1 and thread2 to deadlock ...");
-        awaitBlocked(thread1);
-        awaitBlocked(thread2);
+        awaitBlocked(thread1, reached1);
+        awaitBlocked(thread2, reached2);
```

Comment:
This looks okay but does mean that awaitBlocked is doing two things, maybe it should be rename to awaitTrueAndBlocked. 

An alternative that would remove AtomicReference from the picture is to just have two volatile booleans and have the main thread poll both until true. That would leave the use of awaitBlock unchanged but what you have is okay too.

Reply:
Yes, I actually had that in an initial version. Then I thought it was cleaner to just have all waiting logic in awaitBlocked so I changed it. I renamed `awaitBlocked` to `awaitTrueAndBlocked` for now. But let me know if you prefer the two volatile booleans instead.

Also, we could remove the `CyclicBarrier` and have each thread poll the other’s thread flag now, but given this bug was found due to the barrier I left it as is.

Reply:
Okay, I think what you have is good.